_posts/2021/2021-09-16-5-most-valuable-things-i-learned-in-my-10-plus-year-career.md

```diff
@@ -0,0 +1,124 @@
+- You stay interested in learning new things.
+- You can break the ice with new colleagues more easily.
+- As you become more and more familiar with concepts outside of your knowledge, you become slowly immune against the [Paradox of Expertise](https://www.psychologytoday.com/us/blog/tracking-wonder/201906/the-paradox-expertise){:target="_blank"}.
+- You will get tasks with more responsibility from your employer.
+
+
+
+### 4. Learn about software architecture (how to integrate external systems: file transfer, remote procedure, messaging)
+
+A lot of developers love to code.
+Design patterns will pop up along the developer journey for sure.
+Patterns like MVC, Strategy, Bridge, Adapter, Singleton are almost common knowledge for a developer.
+But there is a lot more to software architecture than this.
+For example, in modern software architecture, you have to integrate a lot of external services instead of implementing them by yourself.
+There are lots of ways to do this (e.g. Webservice based solutions, file-based integrations, and more can be found in this [book](https://www.amazon.de/Patterns-Enterprise-Application-Architecture-Addison-Wesley-ebook/dp/B008OHVDFM/ref=rvi_1/){:target="_blank"}).
+Choosing the right (also technical) solution or pattern can base on many factors.
+One of the biggest influences you have on your decision is based on the purpose and size of the software.
+There are huge differences between [B2B](https://en.wikipedia.org/wiki/Business-to-business){:target="_blank"}, [B2C](https://en.wikipedia.org/wiki/Direct-to-consumer){:target="_blank"}, [B2G](https://en.wikipedia.org/wiki/Business-to-government){:target="_blank"} or [C2C](https://en.wikipedia.org/wiki/Customer_to_customer){:target="_blank"} applications.
+Each will bring its own rules and pitfalls.
+This could be:
+
+- traffic load (B2C)
+- high complexity (B2B)
+- obligation of special laws (B2G)
+- transaction/distribution strategy (C2C)
+
+Documentation of decisions becomes very important and is one pillar of an effective and robust architecture for the distant future.
+If you document your decisions (including arguments for/against other options as well as assumptions), they will make future decisions and your system more robust.
+
+### 5. Be open-minded
+
+While working for several companies over all those years, there is one thing you will always stumble upon - Discussions and disagreements.
+Every developer has a lot of knowledge and thinks their decision or opinion is the right one.
+And the developer tries not to give in as pride is standing in their way.
+
+I would be lying if I said this would not have happened to me.
+At some point, I read an article that was not just about being open-minded, but rather how to be.
+If I need to break it down into short and concise bullet points, I would describe it as:
+
+- Always approach disagreements with curiosity.
+- Always fear being wrong, no matter how confident you are being right.
```

Comment:
"Always fear that you may be wrong, no matter how confident you are that you are right."